File: backend/recipe-service/src/main/java/org/example/recipeservice/entities/Recipe.java
package org.example.recipeservice.entities;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.*;

import jakarta.persistence.*;
import org.example.recipeservice.entities.enums.RecipeStatus;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Recipe {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private String description;

    private String image;

    @Enumerated(EnumType.STRING)
    private RecipeStatus status;

    private LocalDateTime createdAt;

    @JsonIgnoreProperties("recipe")
    @OneToMany(mappedBy = "recipe")
    private List<RecipeSteps> steps;

    @ManyToMany
    @JoinTable(name = "recipe_tags",
            joinColumns = @JoinColumn(name = "recipe_id"),
            inverseJoinColumns = @JoinColumn(name = "tag_id"))
    private List<Tag> tags;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @JsonIgnoreProperties("recipe")
    @OneToMany(mappedBy = "recipe",
            cascade = CascadeType.PERSIST,
            orphanRemoval = true)
    private List<Comment> comments;

    @JsonIgnoreProperties("recipe")
    @OneToMany(mappedBy = "recipe",
            cascade = CascadeType.PERSIST,
            orphanRemoval = true)
    private List<Favorite> favorites;

    @ManyToMany
    @JoinTable(name = "recipe_ingredients",
            joinColumns = @JoinColumn(name = "recipe_id"),
            inverseJoinColumns = @JoinColumn(name = "ingredient_id"))
    private List<Ingredient> ingredients;


}

File: frontend/src/app/core/models/recipe.ts
import { Comment } from "./comment";
import { RecipeStep } from "./recipe-step";
import { Favorite } from "./favorite";
import { Ingredient } from "./ingredient";
import { Tag } from "./tag";
import { Category } from "./category";

export interface RecipeInterface {
	id?: number;
	title?: string;
	description?: string;
	createdAt?: Date | undefined;
	steps?: RecipeStep[];
	comments?: Comment[];
	favorites?: Favorite[];
	ingredients?: Ingredient[];
	tags?: Tag[];
	status?: string;
	category?: Category;
}

export class Recipe implements RecipeInterface {
	constructor(
		public id?: number,
		public title?: string,
		public description?: string,
		public createdAt?: Date | undefined,
		public steps?: RecipeStep[],
		public comments?: Comment[],
		public favorites?: Favorite[],
		public ingredients?: Ingredient[],
		public tags?: Tag[],
		public status?: string,
		public category?: Category,
	) {}
}
